pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -112,18 +126,15 @@ async fn connection_manager_loop_step(
         }
     }
 
-    let id = TenantTimelineId {
-        tenant_id: walreceiver_state.timeline.tenant_id,
-        timeline_id: walreceiver_state.timeline.timeline_id,
-    };
-
     // Subscribe to the broker updates. Stream shares underlying TCP connection
     // with other streams on this client (other connection managers). When
     // object goes out of scope, stream finishes in drop() automatically.
-    let mut broker_subscription = subscribe_for_timeline_updates(broker_client, id).await;
+    let mut broker_subscription =
+        subscribe_for_timeline_updates(broker_client, walreceiver_state.id).await;
     info!("Subscribed for broker timeline updates");
 
     loop {
+        let timeline_read = acquire_timeline_read!(walreceiver_state);
```

Comment:
Why can't we do this in the _one_ place inside this loop that actually needs an `Arc<Timeline>`, i.e., when we get a timeline state update in

```
let new_state = timeline_read.current_state();
```